watson_developer_cloud/discovery_v1.py

```diff
@@ -402,3 +402,262 @@ def query(self, environment_id, collection_id, query_options):
             url='/v1/environments/{0}/collections/{1}/query'.format(
                 environment_id, collection_id),
             params=query_options, accept_json=True)
+
+
+    def delete_training_data(self, environment_id, collection_id):
+        """
+        Clears all training data for this collection.
+        :param environment_id: the ID of your environment
+        :param collection_id: the ID of your collection
+        :return: Code: 204, All training data removed
+        """
+        format_string = '/v1/environments/{0}/collections/{1}/training_data'
+        url_string = format_string.format(environment_id, collection_id)
+        params = {'version': self.version}
+
+        return self.request(method='DELETE',
+                            url=url_string,
+                            params=params)
+
+
+    def list_training_data(self, environment_id, collection_id):
+        """
+        Lists the training data for this collection
+        :param environment_id: the ID of your environment
+        :param collection_id: the ID of your collection
+        :return: Training data for this collection found and returned.
+        """
+        format_string = '/v1/environments/{0}/collections/{1}/training_data'
+        url_string = format_string.format(environment_id, collection_id)
+        params = {'version': self.version}
+
+        return self.request(method='GET',
+                            url=url_string,
+                            params=params,
+                            accept_json=True)
+
+
+    def add_training_data_query(self,
+                                environment_id,
+                                collection_id,
+                                natural_language_query,
+                                filter=None,
+                                examples=None):
+        """
+        Adds a query to the training data for this collection.
+        The query can contain a filter and natural language query
+        :param environment_id: the ID of your environment
+        :param collection_id: the ID of your collection
+        :param natural_language_query:
+        :param filter:
+        :param examples:
+        :return: Training data for this collection found and returned.
+        """
+        format_string = '/v1/environments/{0}/collections/{1}/training_data'
+        url_string = format_string.format(environment_id, collection_id)
+        params = {'version': self.version}
+        data = {'natural_language_query': natural_language_query}
+        if filter:
```

Comment:
Fields in `data` are automatically removed when None, so you don't need these checks.